lib/membrane_rtmp_plugin/rtmp_server.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Membrane.RTMP.Server do
+  @moduledoc """
+  A simple RTMP server, which handles each new incoming connection.
+  """
+  use GenServer
+
+  require Logger
+
+  alias Membrane.RTMP.Server.ClientHandlerBehaviour
+
+  @enforce_keys [:behaviour, :port, :use_ssl?, :listen_options]
+  defstruct @enforce_keys
+
+  @typedoc """
+  Defines options for the RTMP server.
+  """
+  @type t :: %__MODULE__{
+          behaviour: ClientHandlerBehaviour.t(),
+          port: :inet.port_number(),
+          use_ssl?: boolean(),
+          listen_options: any()
+        }
+
+  @type server_identifier :: pid() | atom()
+
+  @doc """
+  Starts the RTMP server.
+  """
+  @spec start_link(server_options :: t(), name :: atom()) :: GenServer.on_start()
+  def start_link(server_options, name \\ nil) do
```

Comment:
Let's move `name` to `server_options`, then we'll be able to pass `{Membrane.RTMP.Server, options}` to a supervisor. Also, the options are usually a keyword list in such cases.